### DISCUSSION TITLE
How to get a node by using the browsename?

### DISCUSSION THREAD

**Discussion by InformatikerImAll** [eclipse-milo/milo#1088] [General]
Is it possible to get a node by using the browsename?  if yes how to get it? right now i am using:

          UaVariableNode node = client.getAddressSpace().getVariableNode(new NodeId(this.namespaceIdx, (int) nodeID));

 i want to use the browsename not the node id toget the node

**Comment by kevinherron** [comment]
No, the browse name is not enough information to identify a Node.

If you know the full _browse path_ to a Node, you can use the TranslateBrowsePathsToNodeId service to get the NodeId, but that still requires you have a starting NodeId and then all the relative browse names that lead to the target Node. It’s meant for programming against types, primarily. Probably not for your use case.
